### PR TITLE
Bug fix

### PR DESCRIPTION
**Pull request by RAdu2005**
Added support for variable arguments on all argument types.
Repaired bug where parser would error on no arguments.
Made it so default is used if argument is not found.
Change Echo command to enforce required-argument-first paradigm.

### CODE REVIEW

File: src/main/java/dev/mlml/command/CommandRegistry.java
Comment (0mlml) [inline src/main/java/dev/mlml/command/CommandRegistry.java:1]: Why

File: src/main/java/dev/mlml/command/Context.java [hunk 1]
```diff
@@ -76,12 +90,15 @@ public boolean parse(Command command) {
...
+            if (i > args.length + i + offset) {
+                if (arg.isRequired()) {
+                    logger.debug("Required argument not found: {}, args.length: {}, offset: {}",
+                                 arg.getName(),
+                                 args.length,
+                                 offset
+                    );
+                    parsedArguments.invalidate();
+                }
                 return true;
```
Comment (0mlml) [inline src/main/java/dev/mlml/command/Context.java:102]: Not entirely convinced this works in the case that it seeks past option arguments, does it fill all of the remaining ones so they don't return null before returning?

File: src/main/java/dev/mlml/command/impl/Echo.java [hunk 2]
```diff
@@ -25,8 +25,8 @@ public Echo() {
                         .get(),
                 new StringArgument.Builder("text")
                         .description("The text to echo")
-                        .require()
                         .isVArgs()
+                        .defaultValue("Echooo")
```
Comment (0mlml) [inline src/main/java/dev/mlml/command/impl/Echo.java:29]: Still think that this should be required, at least for the purposes of testing the parser

File: src/main/java/dev/mlml/command/impl/Help.java [hunk 3]
```diff
@@ -13,16 +13,17 @@
...
         );
     }
 
     @Override
-    public void execute(Context args) {
-        args.getMessage().reply("Helptext placeholder").queue();
+    public void execute(Context ctx) {
+        String commandName = (String) ctx.getArgument("command").getValue();
+
+        System.out.println(commandName);
```
Comment (0mlml) [inline src/main/java/dev/mlml/command/impl/Help.java:27]: Please use slf4j instead of System.out.println

**Review by 0mlml — CHANGES_REQUESTED** [review]

File: src/main/java/dev/mlml/command/impl/Help.java (hunk 3 above)
Comment (RAdu2005) [reply src/main/java/dev/mlml/command/impl/Help.java:27]: That should not have been there.

**Comment by RAdu2005** [comment]
Nvm i found another bug i will fix and create another one

**Comment by RAdu2005** [comment]
Nvm i found another bug i will fix and do another request

**Comment by 0mlml** [comment]
Btw you should've just kept this PR

**Pull request closed without merging**